src/telemetry.rs

```diff
@@ -44,31 +52,63 @@ impl Telemetry {
     /// We are using `impl Subscriber` as return type to avoid having to spell out the actual
     /// type of the returned subscriber, which is indeed quite complex.
     pub fn init(
-        name: String,
+        resource: Resource,
         tracing_settings: TracingSettings,
     ) -> anyhow::Result<(Self, impl Subscriber + Sync + Send)> {
         global::set_text_map_propagator(TraceContextPropagator::default());
 
+        let name = resource.get(semcov::resource::SERVICE_NAME);
+
         let tracer = match tracing_settings.jaeger_collector {
             Some(collector_endpoint) => {
-                opentelemetry_jaeger::new_collector_pipeline()
+                let mut pipeline  = opentelemetry_jaeger::new_collector_pipeline()
                     .with_reqwest()
-                    .with_service_name(&name)
-                    .with_endpoint(collector_endpoint)
-                    .install_batch(runtime::Tokio)?
+                    .with_endpoint(collector_endpoint);
+
+                if let Some(ref name) = name {
+                    pipeline = pipeline.with_service_name(name.to_string());
+                }
+
+                pipeline = pipeline.with_trace_config(
+                    sdktrace::config()
+                        .with_resource(resource)
+                        .with_sampler(sdktrace::Sampler::AlwaysOn),
+                );
+
+                pipeline.install_batch(runtime::Tokio)?
```

Comment:
These lines are the same for all cases. Can we move them out of the `match`? The problem is different types (`CollectorPipeline` vs. `AgentPipeline`). Maybe this macro will help?

```rust
        macro_rules! tracer {
            ($name:ident, $pipeline:expr) => {{
                let mut pipeline = $pipeline;
                if let Some(ref name) = $name {
                    pipeline = pipeline.with_service_name(name.to_string());
                }

                pipeline = pipeline.with_trace_config(
                    sdktrace::config()
                        .with_resource(resource)
                        .with_sampler(sdktrace::Sampler::AlwaysOn),
                );

                pipeline.install_batch(runtime::Tokio)?
            }};
        }
```

Reply:
Fixed